libs/wingc/src/parser.rs

```diff
@@ -1271,13 +1271,23 @@ impl<'s> Parser<'s> {
 		}
 	}
 
-	fn build_class_statement(&self, statement_node: &Node, class_phase: Phase) -> DiagnosticResult<StmtKind> {
+	/// Builds a class field from a class_field node
+	///
+	/// If a class's phase isn't specified, it falls back to the phase of the scope.
+	fn build_class_statement(&self, statement_node: &Node, scope_phase: Phase) -> DiagnosticResult<StmtKind> {
 		let class_modifiers = statement_node.child_by_field_name("modifiers");
-		let class_phase = if self.get_modifier("inflight_specifier", &class_modifiers)?.is_some() {
-			Phase::Inflight
+
+		let class_phase = if let Some(phase) = self.get_phase_specifier(&class_modifiers)? {
+			phase
 		} else {
-			class_phase
+			scope_phase
 		};
+		if class_phase == Phase::Independent {
+			self
+				.with_error::<Node>("Unphased classes are not yet supported", &statement_node)
```

Comment:
Can we use the `UNIMPLEMENTED_GRAMMARS` map for this? Or maybe add a tracking issue like we do for other not yet implemented.

Reply:
Sure thing, I'll add some issue links